Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java

```diff
@@ -52,20 +64,257 @@ public KitodoServiceLoader(Class clazz, String modulePath) {
+    }
+
+    /**
+     * Extracts the module name of the current module by finding the pom.properties
+     * in the folder given by the temporary folder name
+     *
+     * @param temporaryFolderName
+     *            folder name in which the pom.properties file will be searched for
+     *
+     * @return boolean
+     */
+    protected String extractModuleName(String temporaryFolderName) throws IOException {
+        Properties prop = new Properties();
+        InputStream input = null;
+        String moduleName = "";
+        File properties = findFile("pom.properties", temporaryFolderName);
+
+        try {
+            input = new FileInputStream(properties);
+            prop.load(input);
+            return moduleName = prop.getProperty("artifactId");
+
+        } catch (Exception e) {
+            throw e;
+        }
+    }
+
+    /**
+     * This method generates random string
+     *
+     * @return
+     */
+    private String generateRandomString(int randomStringLength){
+        final String charList = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+        StringBuffer randomString = new StringBuffer();
+        for(int i=0; i<randomStringLength; i++){
+            int number = getRandomNumber();
+            randomString.append(charList.charAt(number));
+        }
+        return randomString.toString();
+    }
+
+    /**
+     * This method generates random numbers
+     *
+     * @return int
+     */
+    private int getRandomNumber() {
+        final int lenght = 50;
+        int randomInt;
+        Random randomGenerator = new Random();
+        randomInt = randomGenerator.nextInt(lenght);
+        if (randomInt - 1 == -1) {
+            return randomInt;
+        } else {
+            return randomInt - 1;
+        }
+    }
+
+    /**
+     * Copies extracted frontend files by a given source folder name to the destination Folder
+     * given by a destination folder name
+     *
+     * @param sourceFolder
+     *            copies all extracted frontend files
+     * @param destinationFolder
+     *            jarFile that will be checked for frontend files
+     */
+    private void copyFrontEndFiles(String sourceFolder, String destinationFolder) throws IOException {
+        FileUtils.copyDirectory(new File(sourceFolder), new File(destinationFolder));
+    }
+
+    /**
+     * Checks, whether a passed jarFile has frontend files or not. Returns true, when the jar contains
+     * a folder with the name "resources"
+     *
+     * @param jarPath
+     *            jarFile that will be checked for frontend files
+     * @param destinationPath
+     *            destination path, where the frontend files will be extracted to
+     *
+     */
+    private void extractFrontEndFiles(String jarPath, String destinationPath) throws IOException {
+        File destinationFolder = new File(destinationPath);
+        destinationFolder.mkdir();
+
+        java.util.jar.JarFile jar = new java.util.jar.JarFile(jarPath);
```

Comment:
- Why here and even later in this method full package names are used?
- Please use more meaningful variable names here and even later.